packages/main/src/plugin/kubernetes/contexts-manager-experimental.ts

```diff
@@ -16,12 +16,54 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
-import type { KubeConfig, KubernetesObject } from '@kubernetes/client-node';
+import type { Context, KubernetesObject } from '@kubernetes/client-node';
+import { KubeConfig } from '@kubernetes/client-node';
 
 import type { ContextGeneralState, ResourceName } from '/@api/kubernetes-contexts-states.js';
 
+import { ContextsConnectivityRegistry } from './contexts-connectivity-registry.js';
+import { HealthChecker } from './health-checker.js';
+
+const HEALTH_CHECK_TIMEOUT_MS = 5_000;
+
 export class ContextsManagerExperimental {
-  async update(_kubeconfig: KubeConfig): Promise<void> {}
+  // kubeConfigCheck is used by `update` to check if the received kubeconfig has changed
+  // its value must not be used except for detecting this change
+  // Methods of this class should receive a kubeconfig from `update`
+  #kubeConfigCheck = JSON.stringify(new KubeConfig());
+
+  // Registry to store connectivity state of contexts
+  #contextsConnectivityRegistry = new ContextsConnectivityRegistry();
+
+  // Registry to store health checkers, to be able to abort them when a new update is called
+  #healthCheckers = new Map<string, HealthChecker>();
+
+  async update(kubeconfig: KubeConfig): Promise<void> {
+    const kubeConfigToCheck = JSON.stringify(kubeconfig);
+    if (kubeConfigToCheck === this.#kubeConfigCheck) {
+      // the config did not change since last update, do nothing
+      return;
+    }
+
+    this.disposeAllHealthChecks();
+
+    this.#kubeConfigCheck = kubeConfigToCheck;
+
+    for (const kubeContext of kubeconfig.getContexts()) {
+      const contextName = kubeContext.name;
+      const isolatedConfig = this.getIsolatedKubeConfig(kubeconfig, kubeContext);
+      const healthChecker = new HealthChecker(isolatedConfig);
+      this.#healthCheckers.set(contextName, healthChecker);
+
+      this.#contextsConnectivityRegistry.setState(contextName, true, false);
```

Comment:
IMHO we should not separate the code from the state.

ContextChecker --> define the state for a context (can be merged with HealthChecker)

so instead of having some code here like

```
       const healthChecker = new HealthChecker(isolatedConfig);
      this.#healthCheckers.set(contextName, healthChecker);
      this.#contextsConnectivityRegistry.setState(contextName, true, false);
```

only one object:
```
const contextChecker = new ContextWhateverName(contextName, isolatedConfig);
this.#contextCheckers.set(contextName, contextChecker);
```

Inside the contextChecker object, we know if there is a pending operation (HealthCheck object being instantiated)
last time the reachable has been found, if it's reachable


then in this class, we only need to destroy/recreate the checker object if the config for this context has been changed. 

need an event in the context checker to be notified when a context state is being updated

from that, we can compute what we would like to send to the frontend, can be chunks, full data, whatever, we don't really care as we will just build the data based on all the objects in the tree